xarray/core/dataset.py

```diff
@@ -9304,11 +9305,22 @@ def pad(
             if not var_pad_width:
                 variables[name] = var
             elif name in self.data_vars:
+                if utils.is_dict_like(constant_values):
+                    if name in constant_values.keys():
+                        filtered_constant_values = constant_values[name]
+                    elif not set(var.dims).isdisjoint(constant_values.keys()):
+                        filtered_constant_values = {
+                            k: v for k, v in constant_values.items() if k in var.dims
+                        }
+                    else:
+                        filtered_constant_values = 0
```

Comment:
```suggestion
                        filtered_constant_values = None
```
If I'm not mistaken

Probably a test should cover this as well?

Reply:
If we change that 0 to `None` then there is a discrepancy between what happens if we don't set all `dim` in `constant_variables` (this defaults to padding with 0 and is coded in the `pad`  function from `variable.py`) and if we don't set all `var` (code I added in `pad` function from `dataset.py` which will then pad dimensions with `np.nan`).
I can either change every where which I think might be a breaking change (?) or just leave everything as is.

Reply:
I think this would be worth changing but doesn't need to be in this PR (sorry if I slowed things down by introducing this). Assuming it isn't trivial, adding a TODO with a link to this would be very sufficient

Reply:
No worries, added the TODO :rocket:
Lemme know if there's anything else left to do and thanks for all the comments!